packages/preferences/src/browser/workspace-preference-provider.ts

```diff
@@ -26,9 +26,9 @@ export class WorkspacePreferenceProvider extends AbstractResourcePreferenceProvi
     protected readonly workspaceService: WorkspaceService;
 
     async getUri(): Promise<URI | undefined> {
-        const root = await this.workspaceService.root;
-        if (root) {
-            const rootUri = new URI(root.uri);
+        const workspaceFolder = (await this.workspaceService.workspace).workspaceFolder;
```

Comment:
We could pick any other root to read workspace settings. We should clearly separate workspace and project settings in the future.

Reply:
Yes, that's the plan. Like mentioned in the commit message and PR description, changes to the `preferences` is not included in this PR. for now we use settings in the workspaceFolder as the workspace prefs. And in my next PR, the `preferences` extension will work in the way that, `settings.json` in all root folders are read, and a level of preferences named `FolderPreference` added. prefs at the `FolderPreference` level will only be applied to the folder where they are defined.